### DISCUSSION TITLE
xDrip+ doesn't get data from Libre 2 with OOP2

### DISCUSSION THREAD

**Discussion by JimZam** [NightscoutFoundation/xDrip#2258] [General]
Good morning. I use Libre 2 with patched app to get 12 extra hours out of the sensor (I change in day 10-14) or if the difference between glucometer and xDrip+ is too big (I change to Libre2+OOP2 to be able to calibrate over de -40/+20 limit of the patched app). The problem is xDrip+ doesn't get data with the nightly builds from 27/4 ahead. I always update to the last version, but have to downgrade to 27/4 to make it work with Libre2+OOP2: xDrip+ doesn't work for more than an hour (delete cache, reboot... no way) but it works in a few seconds after downgrade.
Regards and thank you for your help

**Comment by gui-dos** [comment]


**Reply by JimZam** [reply]
Yes, I know. This sensor is 10 days old and no way to make it work with nightly from August de 2nd, but works perfectly with nightly 27th of April and keeps on working if I update now.

**Comment by Navid200** [comment]
@JimZam   Are you saying it works with the version from April 27.  But, does not work with the version from April 28?

**Comment by tzachi-dar** [comment]
@JimZam seems like a very serious problem that you are reporting.
Have you updated to the latest version of oop2? (https://drive.google.com/file/d/17BlRT9aL58hikOIr-MsVs57Lr8seuu5f/view?usp=sharing)
If updating to this version does not work, please install the latest version, add more debug support, scan the sensor and let it run for 15 minutes or so, than send your updated do to tzachi.dar@gmail.com

To add more logging, please enter the value below to "Extra tags for logging"
BgReading:d,jamorham librereceiver:v,LibreOOPAlgorithm:v,jamorham nsemulator:v,DexCollectionService:v

**Comment by JimZam** [comment]
@Navid200 I've not tried every nightly build, just started downgrading with the ones I had in my phone until one of them worked, and it was the one from April 27.
@tzachi-dar It's really strange, it's not the first time it happens to me, last time I asked in gitter. If I downgrade to make it work and then upgrade to last nightly and use Diabox to take bluetooth connection and then xDrip again it works perfectly everytime (I have been scanning with Diabox, then disabling it and using xDrip at least ten times and it works), but first time I try to take bluetooth connection coming from the patched app (and disabling it completely to avoid any kind of interference) it doesn't, no matter what I do. But downgrading to 27/4 works inmediatly...
I'm sorry I can't reproduce the failure until next sensor in a few days :-(

**Comment by nightscout-ricsi** [comment]
I think I had the some problem. I am using a patched LL2 (with xDrip), and if LL2 fails to find the sensor I replace the patched LL2 with OOP2+xDirp. But it does not work for months.

**Comment by tzachi-dar** [comment]
findeszedzo
Can you add more debug logs (see minimal looper) on how to do it, and send
the logs to ***@***.***

**Reply by nightscout-ricsi** [reply]
I will test it again at sensor life endig.

**Reply by nightscout-ricsi** [reply]
I have uploaded log data.

**Comment by JimZam** [comment]
Good morning. It happend to me again last week, I was on holidays with no access to a PC to logcat, all I could get are some screen captures of the event register. I'll send to you right now in case it can be helpful.
Thank you so much

**Reply by nightscout-ricsi** [reply]
I have this issue with OOP2 (Version 3: 8a0ee8f-2022.06.10) long standing, and the only solution is to use an old xDrip version (6th April 2022 - 485662f). I am using [minimalloopers](https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip) recipe.
Problem: that it can not connect per BT, scanning works. I have tried it many times (see log): [one](https://www.dropbox.com/scl/fi/ep7rqnxkh2tgkkmltyp2w/xDrip_error_log_2.xlsx?dl=0&rlkey=b0n4deu14bydus0cexsn3z5xi), [two](https://www.dropbox.com/s/c53nvm9rkl009dt/export20221020-114016.zip?dl=0). I saw a problem many times: some tiny (1 sec) time differences.
See log [on screen](https://www.dropbox.com/s/mvwpfjf5u4e0aj1/Time_problem.jpg?dl=0).

**Reply by PhilTU** [reply]
Never seen all that battery stuff before (LibreBlock, Tomatobattery etc). Mine works good with September xDrip also the "tuned" version https://github.com/jakobheyman/xdripJH

**Reply by nightscout-ricsi** [reply]
Thanks, that version is new for me. I will test it next time, as the OOP2 system is our spare system, if the patched Libre app lost connection per BT.

**Comment by tzachi-dar** [comment]
[findeszedzo](https://github.com/nightscout-ricsi) can you send a copy of your backup file to tzachi dot dar at gmail dot com (I'm writing like this so my mail will not be blocked).

**Comment by kleenkanteen** [comment]
Are you in the EU? I read the patch app only works there

**Reply by nightscout-ricsi** [reply]
Yes, I am in EU. Maybe I do something wrong in the setup. That is left, as a tried it with different phones.
